common/content/statusline.js

```diff
@@ -145,40 +147,74 @@ const StatusLine = Module("statusline", {
 
         if (!bb) return;
 
-        var toggle_off = function () {
+        var hideStatusline = function () {
+            if (!sv.isVisible) {
+                return;
+            }
+
             bb.style.height = '0px';
             bb.style.overflow = 'hidden';
+            sv.contentSeparator = highlight.get('ContentSeparator').value;
             highlight.set('ContentSeparator', 'display: none;');
+            sv.isVisible = false;
         };
 
-        var toggle_on = function () {
+        var showStatusline = function () {
+            if (sv.isVisible) {
+                return;
+            }
+
             bb.style.height = '';
             bb.style.overflow = '';
             highlight.set('ContentSeparator', sv.contentSeparator);
+            sv.isVisible = true;
         };
 
         switch (request) {
-        case sv.TOGGLE:
-            sv.currVisibility = !sv.currVisibility;
-            if (sv.currVisibility) toggle_on();
-            else toggle_off();
-            break;
-        case sv.FULLSCREEN:
-            if (window.fullScreen) {
-                sv.prevVisibility = sv.currVisibility;
-                sv.currVisibility = false;
-                toggle_off();
-            } else {
-                sv.currVisibility = sv.currVisibility || sv.prevVisibility;
-                if (sv.currVisibility) toggle_on();
-            }
-            break;
-        case sv.MODE_ON:
-            if (!sv.currVisibility) toggle_on();
-            break;
-        case sv.MODE_OFF:
-            if (!sv.currVisibility) toggle_off();
-            break;
+            case sv.MODE_AUTO:
+                sv.mode = sv.MODE_AUTO;
+                statusline.setVisibility(sv.EVENT_FULLSCREEN);
+                break;
+            case sv.MODE_ON:
+                sv.mode = sv.MODE_ON;
+                showStatusline();
+                break;
+            case sv.MODE_OFF:
+                sv.mode = sv.MODE_OFF;
+                hideStatusline();
+                break;
+            case sv.EVENT_FULLSCREEN:
+                if (sv.mode != sv.MODE_AUTO) {
```

Comment:
`if (sv.mode != sv.MODE_AUTO || liberator.mode == modes.COMMAND_LINE) {`
This fixes the misbehavior which I commented on.